src/libraries/Native/Unix/System.Native/pal_networkchange.c

```diff
@@ -117,3 +131,44 @@ void SystemNative_ReadEvents(int32_t sock, NetworkChangeEvent onNetworkChange)
         }
     }
 }
+#elif HAVE_RT_MSGHDR
+void SystemNative_ReadEvents(int32_t sock, NetworkChangeEvent onNetworkChange)
+{
+    char buffer[4096];
+    ssize_t count = read(sock, buffer, sizeof(buffer));
+    if (count < 0)
+    {
+        return;
+    }
+
+    struct rt_msghdr msghdr;
+    for (char *ptr = buffer; (ptr + sizeof(struct rt_msghdr)) <= (buffer + count); ptr += msghdr.rtm_msglen)
+    {
+        memcpy(&msghdr, ptr, sizeof(msghdr));
+        if (msghdr.rtm_version != RTM_VERSION)
+        {
+            // version mismatch
+            return;
+        }
+
+        switch (msghdr.rtm_type)
+        {
+            case RTM_NEWADDR:
+                onNetworkChange(sock, AddressAdded);
+                break;
+            case RTM_DELADDR:
+                onNetworkChange(sock, AddressRemoved);
+                break;
+            case RTM_ADD:
+            case RTM_DELETE:
+            case RTM_REDIRECT:
+            {
+                onNetworkChange(sock, AvailabilityChanged);
+                return;
+            }
```

Comment:
Nit: why are the braces needed here but not for the other cases?

Reply:
Removed in https://github.com/dotnet/runtime/pull/2071.